workspaces/src/network/result.rs

```diff
@@ -81,6 +86,60 @@ impl CallExecutionDetails {
         };
         base64::decode(result).map_err(Into::into)
     }
+
+    /// Grab all outcomes after the execution of the transaction. This includes outcomes
+    /// from the transaction and all the receipts it generated.
+    pub fn outcomes(&self) -> Vec<&ExecutionOutcome> {
+        let mut outcomes = vec![&self.transaction];
+        outcomes.extend(&self.receipts);
+        outcomes
+    }
+
+    /// Grab all outcomes after the execution of the transaction. This includes outcomes
+    /// only from receipts generated by this transaction.
+    pub fn receipt_outcomes(&self) -> &Vec<ExecutionOutcome> {
+        &self.receipts
+    }
+
+    /// Grab all outcomes that did not succeed the execution of this transaction. This
+    /// will also include the failures from receipts as well.
+    pub fn failures(&self) -> Vec<&ExecutionOutcome> {
+        let mut failures = Vec::new();
+        if matches!(self.transaction.status, ExecutionStatusView::Failure(_)) {
+            failures.push(&self.transaction);
+        }
+        failures.extend(self.receipt_failures());
```

Comment:
I know optimizing the allocations for this isn't really that useful since it's scripting, but perhaps could use receipt failures length for the vec capacity, or even for the case of the primary transaction not succeeding to just return this to avoid copying and re-allocating?

Honestly not sure if this can be optimized by the compiler

Reply:
Shouldn't this return an Option or Result type? That would avoid the allocation in the negative case.

Reply:
I don't think this function needs to be that optimized. Would probably be better to keep it consistent w/ the other APIs here and return something that is iterable even if we make the allocation. The negative case still works with it just being an empty vec